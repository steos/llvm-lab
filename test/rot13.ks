
	void main()
	{
		int source = 'x';
		int result = rot13(source);
		int back = rot13(result);
		cprintln(source);
		iprintln(source);
		cprintln(result);
		iprintln(result);
		cprintln(back);
		iprintln(back);
	}

	int rot13(int c) {
		int temp = c;
		int cap = temp & 32;
		temp = temp & (~cap);
		if (temp >= 'A' && temp <= 'Z') {
			temp = (temp - 'A' + 13) % 26 + 'A';
		}
		return temp | cap;
	}

	native void cprintln(char);
	
	native void iprintln(int);