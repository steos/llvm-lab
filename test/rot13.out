x
120
k
107
x
120
